.github/workflows/quick.yaml

```
@@ -127,11 +127,18 @@ jobs:
           sudo sed --in-place -E 's/# (deb-src.*updates main)/  \1/g' /etc/apt/sources.list
           sudo apt-get --quiet=2 update
           sudo apt-get --quiet=2 build-dep squid
-          sudo apt-get --quiet=2 install linuxdoc-tools libtool-bin ${{ matrix.compiler.CC }}
+          sudo apt-get --quiet=2 install linuxdoc-tools libtool-bin ${{ matrix.compiler.CC }} ccache
 
       - name: Checkout sources
         uses: actions/checkout@v4
 
+      - name: Setup ccache
+        uses: squid-cache/ccache-action@v1.2.14
+        with:
+          verbose: 2 # default 0
+          append-timestamp: false
+          key: ${{ github.job }}-${{ matrix.os }}-${{ matrix.compiler.CC }}-${{ matrix.layer.nick }}
```

We do not add github.job to log filenames, naturally assuming/implying that no two jobs would test the same os-compiler-layer combo. If possible, let's not add github.job to cache key either (to avoid sending the opposite/contradictory signal).

```suggestion
          key: ${{ matrix.os }}-${{ matrix.compiler.CC }}-${{ matrix.layer.nick }}
```

Same for ccache key in the other file, of course.

---

I do not see that implication. We store the log filenames as they are, and they are distinguished by the job they refer to. Furthermore, these jobs are write-most and rarely read.

The cache does not fulfill the same objective, it is an internal implementation detail which we want to keep separate to avoid thrashing it. The default cache size is 0.5gb, which is enough to keep about 3 run's worth of uncacheable content (e.g. content with a difference in squid.h or autoconf.h). Keeping it separate per job is a way to maximise its effectiveness, letting github's eviction policies take care of not overflowing it.

In other words, I'd leave github.job in here; if you feel that there is a risk of confusion (I don't, but I accept your opinion) we can add a comment.

---

The actual key used to restore the cache from a sample job is
`ccache-build-tests-ubuntu-22.04-gcc-default-2024-11-05T16:19:52.056Z`

The key question is: what is it namespaced to? The short answer is, it's hard to tell.
The code restoring the cache is at https://github.com/actions/cache/blob/main/src/saveImpl.ts , but it has so many dependencies into other workflows that it would take a fair amount of time to trace.

Sensible namespace scopes could be:
- the github account
- the repository
- the branch
- the workflow (file)
- the job

Documentation at https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows seems to imply (but never states) that the scope is the branch (with additional branches added to the scope when the action is evaluated for a PR)

I'd rather approach this the other way around: can anyone demonstrate that having the job name in the key will hurt, besides the confusion issue you mention which can be trivially solved by a comment?
Is the potential harm of creating confusion in a scope that's unlikely to ever be considered again by anyone else even worth the time we are spending in the discussion?

---

> I'd rather approach this the other way around

I understand the positives behind the other way around, but have to insist that, given the current resources and responsibilities, we justify additions/changes (rather than adding/changing unless proven wrong).

> can anyone demonstrate that having the job name in the key will hurt, besides the confusion issue you mention which can be trivially solved by a comment? 

There is no confusion that I know of. Adding a comment does not solve the inconsistency (for the lack of a better word) problem that this change request has identified.

> Is the potential harm of creating confusion in a scope that's unlikely to ever be considered again by anyone else even worth the time we are spending in the discussion?

We are definitely wasting time here, but that waste started when the inconsistency in question was introduced, not when I dared to flag it. Accepting my recommendation that would address it would take a few minutes. If that recommendation turns out to be wrong, we will learn something useful about these workflows _and_ use that knowledge to improve them (as opposed to adding a comment explaining that inconsistencies are there "just in case" or "because this code works" or whatever that excuse is going to be).

> The key question is: what is it namespaced to?

GitHub [Cache Action](https://github.com/actions/cache/tree/main) namespace is a GitHub repository AFAICT. There is also a cache entry "versioning" mechanism that prevents (by default) cross-OS sharing of cache entries. That default protection can be disabled via enableCrossOsArchive.

FWIW, the above is _not_ the key question IMO. The key question is: Can two distinct os-compiler-layer builds co-exist across our current workflows for a given git commit? If the answer (to a meaningful interpretation of that question) is yes, then we need to fix our existing and PR workflows that assume the opposite answer! If the answer is no, then there is no need for this PR to add a job name to the existing, "working", and "reasonable" os-compiler-layer build key.

For example, if we were to test _both_ OpenSSL and GnuTLS builds for the same os-compiler-layer, then os-compiler-layer key would stop working.